### PR TITLE
chore(dep): Bump gateway api to the lastest version

### PR DESCRIPTION
**Pull request by sayboras**
Please refer to individual  commit for more details.

### CODE REVIEW

File: .github/workflows/conformance-gateway-api.yaml [hunk 1]
```diff
@@ -193,14 +190,15 @@ jobs:
 
       - name: Install Gateway API CRDs
         run: |
+          gateway_api_version=$(grep "sigs.k8s.io/gateway-api" go.mod | awk -F'-' '{print (NF>1)?$NF:$0}')
```
Comment (youngnick) [inline .github/workflows/conformance-gateway-api.yaml:193]: thumbsup

**Review by youngnick — APPROVED** [review]
LGTM, nice work!

**Review by viktor-kurchenko — APPROVED** [review]
Cool!

**Review by marseel — APPROVED** [review]

**Pull request merged**